### DISCUSSION TITLE
Future support for multi-zones (or module federation)

### DISCUSSION THREAD

**Discussion by maxsteenbergen** [vercel/community#3768] [Feature Question]
### Question

For our application's new frontend, we are investigating and leaning towards a micro-frontend based architecture.
With the Pages router, we might be able to leverage the multi-zones feature. However, any mention of multi-zones has been removed from the docs related to the App router.
Are multi-zones to be deprecated and if so, what should we use as a replacement?

Related: how does Vercel/Next.JS look at Module Federation, and does or will it offer any similar functionality?

**Comment by amyegan** [comment]
Hey @maxsteenbergen. Multi-zone is essentially just a combination rewrites and base paths. You can do this with [rewrites](https://nextjs.org/docs/app/api-reference/next-config-js/rewrites) and [basePath](https://nextjs.org/docs/app/api-reference/next-config-js/basePath) in `next.config.js`. I put together a quick [example using the App router](https://github.com/amyegan/multi-zone-with-app-router), which is based on the [existing example](https://github.com/vercel/next.js/tree/canary/examples/with-zones) which uses Pages. A [monorepo](https://vercel.com/blog/monorepos) is also worth considering if you haven't already.

**Reply by maxsteenbergen** [reply]
@amyegan thanks! So to paraphrase: essentially multi-zones is still supported, just not named as such anymore.

What about Module Federation? What is Vercel's stance or alternative to that, if any?

**Reply by tonyabracadabra** [reply]
Can we mix astro with nextjs through rewrites?

**Reply by amyegan** [reply]
Yes, you can have multiple projects with different frameworks and make them appear as one by using rewrites.

- [Rewrites on Vercel](https://vercel.com/docs/edge-network/rewrites#)
- [next.config.js Options: rewrites](https://nextjs.org/docs/pages/api-reference/next-config-js/rewrites)
